gtests/test_nopython_transform.cpp

```diff
@@ -0,0 +1,111 @@
+#include <modmesh/modmesh.hpp>
+#include <modmesh/transform/transform.hpp>
+#include <random>
+#include <gtest/gtest.h>
+
+#ifdef Py_PYTHON_H
+#error "Python.h should not be included."
+#endif
+
+template <typename T>
+class ParsevalTest : public ::testing::Test
+{
+protected:
+    const size_t VN = 1024;
+
+    modmesh::SimpleArray<modmesh::Complex<T>> signal{
+        modmesh::small_vector<size_t>{VN}, modmesh::Complex<T>(0.0, 0.0)};
+    modmesh::SimpleArray<modmesh::Complex<T>> out{
+        modmesh::small_vector<size_t>{VN}, modmesh::Complex<T>(0.0, 0.0)};
+
+    // Set up the test fixture: generate the signal once
+    void SetUp() override
+    {
+        std::mt19937 rng{std::random_device{}()};
+        std::uniform_real_distribution<T> val_dist{-1.0, 1.0};
+        for (unsigned int i = 0; i < VN; ++i)
+        {
+            T val = val_dist(rng);
+            signal[i] = modmesh::Complex<T>(val, 0.0);
+        }
+    }
+
+    // Function to verify energy conservation (Parseval's theorem)
+    void verify_parseval()
+    {
+        T psd_sig = 0.0, psd_out = 0.0;
+
+        for (unsigned int i = 0; i < VN; ++i)
+        {
+            psd_sig += modmesh::norm<modmesh::Complex, T>(signal[i]);
+            psd_out += modmesh::norm<modmesh::Complex, T>(out[i]);
+        }
+
+        // Expect the total energy in the time and frequency domains to be equal
+        EXPECT_NEAR(psd_sig, psd_out, (std::is_same<T, float>::value ? (T)1e-3 : (T)1e-10));
```

Comment:
Actually, I’m not sure how to decide the tolerance for the FFT in both `float` and `double`.

I also noticed that when the FFT uses `float`, the accumulation error increases rapidly. While I know that float loses precision, this makes me worry if there might be an issue with the implementation.

Reply:
Leave a comment for what you found and worry about it later.